aten/src/ATen/native/cuda/AveragePool2d.cu

```diff
@@ -427,7 +413,7 @@ Tensor& avg_pool2d_backward_out_cuda_template(
 
   const int32_t count = safe_downcast<int32_t, int64_t>(input.numel());
   const uint32_t num_threads = std::min(at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, 1024);
-  const uint32_t num_blocks = cuda::ATenCeilDiv<uint32_t>(count, num_threads);
+  const uint32_t num_blocks = cuda::ATenCeilDiv<uint32_t>(count ? count : 1, num_threads);
 
   bool use_divisor = divisor_override.has_value();
   const auto divisor_override_value = use_divisor ? divisor_override.value() : 0;
```

Comment:
Also, don't launch kernel if backward `numel()` is zero.

Reply:
So that would put an extra if-condition around the kernel, which this change will make unnecessary. I had put that in the previous commit to which you replied that there are too many if-conditions in the code. Should I put that back?

Reply:
I would strongly suggest that we do not launch unnecessary cuda kernels if possible, since this could be expensive. cc @ngimel